src/sql/workbench/contrib/notebook/browser/notebookActions.ts

```diff
@@ -308,11 +308,31 @@ export class RunParametersAction extends TooltipFromLabelAction {
 		const editor = this._notebookService.findNotebookEditor(context);
 		// Set defaultParameters to the parameter values in parameter cell
 		let defaultParameters = new Map<string, string>();
+		let emptyParameterCell: boolean;
 		editor.cells.forEach(cell => {
 			if (cell.isParameter) {
+				// Check if cell is empty or contains just whitespace
+				for (let x of cell.source) {
```

Comment:
This doesn't work as expected if source is a string - it'll iterate over each character in the string and test that separately. So if that string has a space in it then it'll think it's empty.